pkg/kernel/networkservice/inject/common.go

```diff
@@ -61,16 +79,57 @@ func move(logger log.Logger, conn *networkservice.Connection, isMoveBack bool) e
 	}
 	defer func() { _ = targetNetNS.Close() }()
 
+	vfConfig := vfconfig.Config(ctx)
 	ifName := mech.GetInterfaceName()
 	if !isMoveBack {
-		err = moveInterfaceToAnotherNamespace(ifName, curNetNS, curNetNS, targetNetNS)
+		err = moveToContNetNS(vfConfig, ifName, curNetNS, targetNetNS)
 	} else {
-		err = moveInterfaceToAnotherNamespace(ifName, curNetNS, targetNetNS, curNetNS)
+		err = moveToHostNetNS(vfConfig, ifName, curNetNS, targetNetNS)
 	}
 	if err != nil {
-		logger.Warnf("failed to move network interface %s into the target namespace for connection %s", ifName, conn.GetId())
+		// link may not be available at this stage (might be deleted in previous chain element itself) for veth case
+		if strings.Contains(err.Error(), "Link not found") {
+			return nil
+		}
 		return err
 	}
-	logger.Infof("moved network interface %s into the target namespace for connection %s", ifName, conn.GetId())
 	return nil
 }
+
+func moveToContNetNS(vfConfig *vfconfig.VFConfig, ifName string, curNetNS, targetNetNS netns.NsHandle) (err error) {
+	link, _ := kernellink.FindHostDevice("", ifName, targetNetNS)
+	if link != nil {
+		return
+	}
+	if vfConfig != nil && vfConfig.VFInterfaceName != ifName {
+		err = moveInterfaceToAnotherNamespace(vfConfig.VFInterfaceName, curNetNS, curNetNS, targetNetNS)
+		if err == nil {
+			err = renameInterface(vfConfig.VFInterfaceName, ifName, curNetNS, targetNetNS)
+		}
+	} else {
+		err = moveInterfaceToAnotherNamespace(ifName, curNetNS, curNetNS, targetNetNS)
+	}
+	return
+}
+
+func moveToHostNetNS(vfConfig *vfconfig.VFConfig, ifName string, curNetNS, targetNetNS netns.NsHandle) (err error) {
+	if vfConfig != nil && vfConfig.VFInterfaceName != ifName {
+		link, _ := kernellink.FindHostDevice("", vfConfig.VFInterfaceName, curNetNS)
```

Comment:
If interface was renamed, we need to find it by `ifName` and so if we are moving it back, we probably need to find it in `targetNetNS`.

Reply:
This method gets executed only for sriov device move back scenario and renaming would happen only at line 123. 
This check is done to make things to be idempotent and also to find out if sriov device is accidentally moved out to hostNetNS without renaming the interface (in case of container kill), its not appropriate to rely on `ifName` here as multiple NSCs use same interface name. so we must also search using `kernellink.FindHostDevice` using VFs pci address to restore interface original name. will revisit here once this [PR](https://github.com/networkservicemesh/sdk-kernel/pull/307) is merged.